cmd/blocktx.go

```diff
@@ -94,17 +81,50 @@ func StartBlockTx(logger *slog.Logger) (func(), error) {
 		return nil, err
 	}
 
-	peerHandler, err := blocktx.NewPeerHandler(logger, blockStore, startingBlockHeight, peerURLs, network,
+	peerHandler, err := blocktx.NewPeerHandler(logger, blockStore,
 		blocktx.WithRetentionDays(recordRetentionDays),
 		blocktx.WithTxChan(txChannel),
 		blocktx.WithRegisterTxsInterval(registerTxInterval),
 		blocktx.WithMessageQueueClient(mqClient))
-
 	if err != nil {
 		return nil, err
 	}
 
-	blockTxServer := blocktx.NewServer(blockStore, logger, peerHandler)
+	peerHandler.Start()
+
+	pm := p2p.NewPeerManager(logger, network, p2p.WithExcessiveBlockSize(maximumBlockSize))
+
+	peerSettings, err := config.GetPeerSettings()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get peer settings: %v", err)
+	}
+
+	peerURLs := make([]string, len(peerSettings))
+	for i, peerSetting := range peerSettings {
+		peerUrl, err := peerSetting.GetP2PUrl()
+		if err != nil {
+			return nil, fmt.Errorf("error getting peer url: %v", err)
+		}
+		peerURLs[i] = peerUrl
+	}
+	peers := make([]p2p.PeerI, len(peerURLs))
+
+	for i, peerURL := range peerURLs {
+		peer, err := p2p.NewPeer(logger, peerURL, peerHandler, network, p2p.WithMaximumMessageSize(maximumBlockSize))
+		if err != nil {
+			return nil, fmt.Errorf("error creating peer %s: %v", peerURL, err)
+		}
+		err = pm.AddPeer(peer)
+		if err != nil {
+			return nil, fmt.Errorf("error adding peer: %v", err)
+		}
+
+		peers[i] = peer
```

Comment:
Would it be beneficial in the future to have the `peers` array saved in peerHandler? For instance I needed it to check that each peer was healthy and connected.

Reply:
The reason I took `peers` out of `peerHandler` is that `peerHandler` is passed as a parameter to each peer when it is created. 
https://github.com/bitcoin-sv/arc/blob/fb75bb464d40f8ad3084175107ffc3c29369627e/cmd/blocktx.go#L113

So I found it strange if `peerHandler` is inside each `peer` and in turn all `peers` are inside `peerHandler`.

The way it is done now is that `peers` is passed to the server where they can be used for the health check instead of going via the `peerHandler`.

https://github.com/bitcoin-sv/arc/blob/fb75bb464d40f8ad3084175107ffc3c29369627e/blocktx/health_check.go#L37

Does that make sense to you?

Reply:
yeap